src/coreclr/vm/assembly.cpp

```diff
@@ -643,13 +662,13 @@ Assembly *Assembly::CreateDynamic(AppDomain *pDomain, CreateDynamicAssemblyArgs
         {
             GCX_PREEMP();
             // Assembly::Create will call SuppressRelease on the NewHolder that holds the LoaderAllocator when it transfers ownership
-            pAssem = Assembly::Create(pDomain, pFile, pDomainAssembly->GetDebuggerInfoBits(), args->access & ASSEMBLY_ACCESS_COLLECT ? TRUE : FALSE, pamTracker, pLoaderAllocator);
+            pAssem = Assembly::Create(pDomain, pFile, pDomainAssembly->GetDebuggerInfoBits(), ((args->access & ASSEMBLY_ACCESS_COLLECT) | pLoaderAllocator->IsCollectible()) ? TRUE : FALSE, pamTracker, pLoaderAllocator);
```

Comment:
Can we simply use `pLoaderAllocator->IsCollectible` - I think it should be always true if `ASSEMBLY_ACCESS_COLLECT` was used. And I think it's wrong if collectible assembly is created on non-collectible allocator.

Reply:
Left only check `pLoaderAllocator-> IsCollectible`